src/module-rebuilder.ts

```diff
@@ -151,6 +151,10 @@ export class ModuleRebuilder {
       return true;
     }
 
+    if (this.rebuilder.platform !== process.platform) {
+      throw new Error("OS Limitation - It is not possible to cross-compile native modules from source")
+    }
```

Comment:
I took a look at that approach, but the reason I didn't was because then it seemed like electron-rebuild would try to rebuild every native module and throw an individual error each time
https://github.com/electron/rebuild/blob/cb372cdf09d9ca3ac8f1231608e4ebedddabe51c/src/rebuild.ts#L237-L241

Instead, I felt that putting the check at the beginning would be an early exit from the flow

Reply:
Following up here, I ended up moving the logic into `rebuildModule` as you suggested to simplify/isolate behaviors for unit testing specific rebuild/prebuild files as opposed to pulling the entire `ModuleRebuilder` into `module-type-node-gyp.ts` test case